code/modules/research/xenobiology/xenobiology.dm

```diff
@@ -447,6 +447,46 @@
 	if(loc == usr && loc.Adjacent(over_object))
 		afterattack(over_object, usr, TRUE)
 
+/obj/item/slimepotion/oil_slick
+	name = "slime oil potion"
+	desc = "A potent chemical mix that will remove the slowdown from any item by reducing friction. Doesn't mix well with water."
+	icon = 'icons/obj/chemical.dmi'
+	icon_state = "bottle3"
+	origin_tech = "biotech=5"
+
+/obj/item/slimepotion/oil_slick/afterattack(obj/O, mob/user, proximity_flag)
+	if(!proximity_flag)
+		return
+	..()
+	if(!istype(O))
+		to_chat(user, "<span class='warning'>The potion can only be used on items or vehicles!</span>")
+	if(isitem(O))
+		var/obj/item/I = O
+		if(I.slowdown <= 0)
+			to_chat(user, "<span class='warning'>[I] can't be made any faster!</span>")
+			return ..()
```

Comment:
I don't think there should ever be more than one afterattack parent call. So, not sure about these `return ..()`s if the code already does `..()` above. May just need to be `return`.

Reply:
can do